lib/realtime.js

```diff
@@ -625,31 +625,42 @@ function updateUserData (socket, user) {
 function ifMayEdit (socket, callback) {
   const noteId = socket.noteId
   if (!noteId || !notes[noteId]) return
-  const note = notes[noteId]
-  let mayEdit = true
-  switch (note.permission) {
-    case 'freely':
-      // not blocking anyone
-      break
-    case 'editable': case 'limited':
-      // only login user can change
-      if (!socket.request.user || !socket.request.user.logged_in) { mayEdit = false }
-      break
-    case 'locked': case 'private': case 'protected':
-      // only owner can change
-      if (!note.owner || note.owner !== socket.request.user.id) { mayEdit = false }
-      break
-  }
-  // if user may edit and this is a text operation
-  if (socket.origin === 'operation' && mayEdit) {
-    // save for the last change user id
-    if (socket.request.user && socket.request.user.logged_in) {
-      note.lastchangeuser = socket.request.user.id
-    } else {
-      note.lastchangeuser = null
+  models.Note.findOne({
+    where: {
+      id: noteId
+    },
+    attributes: ['id', 'permission']
+  }).then(function (note) {
+    let mayEdit = true
```

Comment:
if this is edited anyway: I'd start with `mayEdit = false` and then allow it when criteria are met (currently it's the other way round). This would remove a ton of negations down in the case statements, making it clearer overall.

Reply:
I hope the new commit fixes this to your liking :)

Reply:
In my opinion the new commit is much more readable. Thanks!